includes/wikia/models/DesignSystemGlobalFooterModel.class.php

```diff
@@ -545,4 +332,91 @@ private function getHref( $hrefKey ) {
 	private function getSocialHrefs() {
 		return DesignSystemSharedLinks::getInstance()->getSocialHrefs( $this->lang );
 	}
+
+	private function getLinkTextObjectList( String $section, Array $linkKeys ) {
+		return array_map(
+			function ( $link ) use ( $section ) {
+				return $this->getLinkTextTranslatableObject(
+					'global-footer-' . $section . '-link-' . $link,
+					$link,
+					$section . '.' . $link
+				);
+			},
+			$linkKeys
+		);
+	}
+
+	private function getLinkTextTranslatableObject( $titleKey, $hrefKey, $trackingLabel ) {
```

Comment:
Yes, I was thinking about that, but firstly I wanted to check if this is good direction. And extracting those methods to separate classes makes even more sense if we'd like to apply those changes also to Global navigation